### DISCUSSION TITLE
Change detection through asset handles?

### DISCUSSION THREAD

**Discussion by torsteingrindvik** [bevyengine/bevy#11631] [Q&A] [answered]
Today I tried something along the lines of 

```rust
q: Query<(&mut Transform, &Handle<MyAsset>), Changed<Handle<MyAsset>>
```

hoping that hot-reloading updates on the asset would trigger the change detection filter.

It didn't work. Is this expected since the asset is behind a handle? Is there an idiomatic way of doing this?

**Comment by mockersf** [comment, marked as answer]
Your filter would match if the handle itself changed on this entity, not if the assets pointed to by the handle changed.

When an asset changed, you'll get an event [`AssetEvent`](https://docs.rs/bevy/latest/bevy/asset/enum.AssetEvent.html), up to you to keep a map handle <-> Vec<EntityId>

**Reply by torsteingrindvik** [reply]
I was hoping there wasn't a need as a user to do this mapping.

I saw now that in https://github.com/bevyengine/bevy/issues/11266 : 

> Change detection for assets would just be regular component change detection.

So I suppose I can observe that issue.
